samples/msal-node-samples/EndToEndTestApp/app/test/auth-code-aad/auth-code-aad.spec.ts

```diff
@@ -0,0 +1,156 @@
+            testName = "authCodeAcquireToken";
+            screenshot = new Screenshot(`${SCREENSHOT_BASE_FOLDER_NAME}/${testName}/${environment}`);
+            await NodeCacheTestUtils.resetCache(TEST_CACHE_LOCATION);
+        });
+
+        beforeEach(async () => {
+            context = await browser.createIncognitoBrowserContext();
+            page = await context.newPage();
+            page.on('dialog', async dialog => {
+                console.log(dialog.message());
+                await dialog.dismiss();
+            });
+            page.setDefaultNavigationTimeout(0);
+        });
+
+        afterEach(async () => {
+            await page.close();
+            await context.close();
+            await NodeCacheTestUtils.resetCache(TEST_CACHE_LOCATION);
+        });
+
+        it("Performs acquire token", async () => {
+            await page.goto(HOME_ROUTE);
+            await enterCredentials(page, screenshot, username, accountPwd);
+
+            const cachedTokens = await NodeCacheTestUtils.waitForTokens(TEST_CACHE_LOCATION, 2000);
+            expect(cachedTokens.accessTokens.length).toBe(1);
+            expect(cachedTokens.idTokens.length).toBe(1);
+            expect(cachedTokens.refreshTokens.length).toBe(1);
+         });
+
+        it("Performs acquire token with prompt = 'login'", async () => {
+            await page.goto(`${HOME_ROUTE}/?prompt=login`);
+            await enterCredentials(page, screenshot, username, accountPwd);
+
+            const cachedTokens = await NodeCacheTestUtils.waitForTokens(TEST_CACHE_LOCATION, 2000);
+            expect(cachedTokens.accessTokens.length).toBe(1);
+            expect(cachedTokens.idTokens.length).toBe(1);
+            expect(cachedTokens.refreshTokens.length).toBe(1);
+        });
+
+        it("Performs acquire token with prompt = 'consent'", async () => {
+            await page.goto(`${HOME_ROUTE}/?prompt=consent`);
+            await enterCredentialsWithConsent(page, screenshot, username, accountPwd);
+
+            const cachedTokens = await NodeCacheTestUtils.waitForTokens(TEST_CACHE_LOCATION, 2000);
+            expect(cachedTokens.accessTokens.length).toBe(1);
+            expect(cachedTokens.idTokens.length).toBe(1);
+            expect(cachedTokens.refreshTokens.length).toBe(1);
+        });
+
+        it("Performs acquire token with prompt = 'none'", async () => {
+            // First log the user in first
+            await page.goto(`${HOME_ROUTE}/?prompt=login`);
+            await enterCredentials(page, screenshot, username, accountPwd);
+            await NodeCacheTestUtils.waitForTokens(TEST_CACHE_LOCATION, 2000);
+            
+            // Reset the cache to prepare for the second login
+            await NodeCacheTestUtils.resetCache(TEST_CACHE_LOCATION);
+
+            // Login without a prompt 
+            await page.goto(`${HOME_ROUTE}/?prompt=none`);
+            await new Promise(resolve => { setTimeout(() => { resolve(true) }, 4000)})
```

Comment:
Is there something more meaningful you can wait for here instead of waiting a fixed 4 seconds? This will slow down the test suite a lot if used too much.

Reply:
@tnorling I added this to wait for the browser to complete it's various redirects. Any idea of how to better manage this?

Reply:
My bad was talking about a different part of the code base.